NuRadioReco/modules/io/event_parser_factory.py

```diff
@@ -162,18 +170,26 @@ def iter_events_2_2(self):
                     object_type = int.from_bytes(object_type_hex, 'little')
                     bytes_to_read_hex = self._get_file(self._current_file_id).read(6)
                     bytes_to_read = int.from_bytes(bytes_to_read_hex, 'little')
+                    current_byte = 24
```

Comment:
are you sure that this is what you want?

Reply:
I tested it and it works.
Unfortunately, during tests, I noticed that the io cannot handle stacked loops because self_current_file_id will change in the inner loop and be at the last file when the outer loop continues.
I honestly think it is better to rewrite the NuRadioRecoio, because the way it is right now is very messy and I think there is a simpler way to do tthis.

Reply:
good that you found that. Yes this is tricky... 
Agreed that it might be easier to rewrite the io, but I can't really tell before looking deeply into it. If you decide to do so, please start with a new "NuRadioRecoIO2" file (or similar) so that we can cross check both versions.

Reply:
My plan is to scan the file when it is opened and make a lists of event IDs and their location in the file. The iterator would then iterate over this list instead of going through the bytes in the file directly. That would get rid of a lot of the byte counting that we are doing right now.
But this is going to be its own pull request. This one is already going to be large enough as it is.

Reply:
There is one downside of this approach. It takes a long time to parse (multiple) large files. This why the "parse header" is optional. If one just uses the event Reader, it just starts reading the file byte by byte. 

But we could probably create the event to file_id/byte list dynamically and only read the file on demand. Please keep this in mind when you start implementing it.